xilem/examples/http_cats/main.rs

```diff
@@ -29,36 +41,80 @@ impl HttpCats {
                 .collect::<Vec<_>>(),
         )));
-        let info_area = if let Some(selected_code) = self.selected_code {
+        let (info_area, worker_value) = if let Some(selected_code) = self.selected_code {
             if let Some(selected_status) =
                 self.statuses.iter_mut().find(|it| it.code == selected_code)
             {
-                OneOf3::A(selected_status.details_view())
+                // If we haven't requested the image yet, make sure we do so.
+                let value = match selected_status.image {
+                    ImageState::NotRequested => {
+                        // TODO: Should a view_function be editing `self`?
+                        // This feels too imperative.
+                        selected_status.image = ImageState::Pending;
+                        Some(selected_code)
+                    }
+                    // If the image is pending, that means that worker already knows about it.
+                    // We don't set the requested code to `selected_code` here because we could have been on
+                    // a different view in-between, so we don't want to request the same image twice.
+                    ImageState::Pending => None,
+                    ImageState::Available(_) => None,
+                };
+                (OneOf3::A(selected_status.details_view()), value)
             } else {
-                OneOf3::B(
-                    prose(format!(
-                        "Status code {selected_code} selected, but this was not found."
-                    ))
-                    .alignment(TextAlignment::Middle)
-                    .brush(Color::YELLOW),
+                (
+                    OneOf3::B(
+                        prose(format!(
+                            "Status code {selected_code} selected, but this was not found."
+                        ))
+                        .alignment(TextAlignment::Middle)
+                        .brush(Color::YELLOW),
+                    ),
+                    None,
                 )
             }
         } else {
-            OneOf3::C(
-                prose("No selection yet made. Select an item from the sidebar to continue.")
-                    .alignment(TextAlignment::Middle),
+            (
+                OneOf3::C(
+                    prose("No selection yet made. Select an item from the sidebar to continue.")
+                        .alignment(TextAlignment::Middle),
+                ),
+                None,
             )
         };
 
-        flex((
-            // Add padding to the top for Android. Still a horrible hack
-            FlexSpacer::Fixed(40.),
-            flex((left_column.flex(1.), info_area.flex(1.)))
-                .direction(Axis::Horizontal)
-                .must_fill_major_axis(true)
-                .flex(1.),
-        ))
-        .must_fill_major_axis(true)
+        // TODO: Should `web_image` be a built-in component?
```

Comment:
Yes. Especially if we implement a form of caching.

Reply:
Agreed - caching absolutely is the giant question mark on that aspect, which is why I'm building it "externally" for now.